src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerCommitImageFunctionalTest.groovy

```diff
@@ -0,0 +1,54 @@
+package com.bmuschko.gradle.docker.tasks.image
+
+import com.bmuschko.gradle.docker.AbstractGroovyDslFunctionalTest
+import org.gradle.testkit.runner.BuildResult
+
+class DockerCommitImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
+
+    def "can commit image"() {
+        given:
+        String commitImage = """
+            task commitImage(type: DockerCommitImage) {
+                dependsOn startContainer
+                targetContainerId createContainer.getContainerId()
+                tag = "myimage:latest"
+            }
+        """
+
+        buildFile << containerStart(commitImage)
+
+        when:
+        BuildResult result = build('commitImage')
+
+        then:
+        result.output.contains("Commiting image for container")
+    }
+
+    static String containerStart(containerCommitImageExecutionTask) {
+        """
+            import com.bmuschko.gradle.docker.tasks.container.DockerCreateContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerStartContainer
+            import com.bmuschko.gradle.docker.tasks.image.DockerCommitImage
+            import com.bmuschko.gradle.docker.tasks.image.DockerPullImage
+
+            task pullImage(type: DockerPullImage) {
+                repository = '$TEST_IMAGE'
+                tag = '$TEST_IMAGE_TAG'
+            }
+
+            task createContainer(type: DockerCreateContainer) {
+                dependsOn pullImage
+                targetImageId pullImage.getImageId()
+                autoRemove = true
+                entrypoint = ['tail', '-f', '/dev/null']
+            }
+
+            task startContainer(type: DockerStartContainer) {
+                dependsOn createContainer
+                targetContainerId createContainer.getContainerId()
+            }
+            
+            ${containerCommitImageExecutionTask}
```

Comment:
This test case is missing a task that stops the container. If we don't stop the container then we'll end up with stranded containers whenever the functional test is run. Could you please add it as a `finalizedBy` task? You will find some examples in other test classes.

Reply:
Thanks for the feedback. I added image and container clean up in the test. Let me know how it looks.